cpp/bench/ann/src/common/benchmark.hpp

```diff
@@ -319,22 +367,41 @@ void register_build(std::shared_ptr<const Dataset<T>> dataset,
     auto* b = ::benchmark::RegisterBenchmark(
       index.name + suf, bench_build<T>, dataset, index, force_overwrite);
     b->Unit(benchmark::kSecond);
+    b->MeasureProcessCPUTime();
     b->UseRealTime();
   }
 }
 
 template <typename T>
 void register_search(std::shared_ptr<const Dataset<T>> dataset,
-                     std::vector<Configuration::Index> indices)
+                     std::vector<Configuration::Index> indices,
+                     Objective metric_objective)
 {
   for (auto index : indices) {
     for (std::size_t i = 0; i < index.search_params.size(); i++) {
       auto suf = static_cast<std::string>(index.search_params[i]["override_suffix"]);
       index.search_params[i].erase("override_suffix");
-      auto* b =
-        ::benchmark::RegisterBenchmark(index.name + suf, bench_search<T>, dataset, index, i);
-      b->Unit(benchmark::kMillisecond);
-      b->UseRealTime();
+
+      int max_threads =
+        metric_objective == Objective::THROUGHPUT ? std::thread::hardware_concurrency() : 1;
+
+      auto* b = ::benchmark::RegisterBenchmark(
+                  index.name + suf, bench_search<T>, index, i, dataset, metric_objective)
+                  ->Unit(benchmark::kMillisecond)
+                  ->ThreadRange(1, max_threads)
```

Comment:
We can add as a follow-on if this causes problems. I almost added that feature but i don't know that this will often go above the max threads available to the hardware and it does help to have it increase from a single thread by powers of 2 because (1: it doesn't add that many additional experiments, and 2: it quickly shows the saturation point when latencies no longer decrease with increased parallelism). I also like showing the number of threads even for the "latency" mode because it makes the naming of the experiments consistent.